clients/ValkeyClients.md

```diff
@@ -0,0 +1,218 @@
+
+## Mandatory Features Overview
+
+1. **Cluster Support** - The ability to operate in a clustered environment, where the data is distributed across multiple shards. Cluster support is essential for applications that require high  scalability.
+
+2. **TLS/SSL Support** - The capability to establish secure connections using TLS/SSL, which encrypts the data transmitted between the client and the server. This is a critical feature for applications that require data privacy and protection against eavesdropping.
+
+## Advanced Features Overview
+
+1. **Read from Replica** - The ability to read data from a replica node, which can be useful for load balancing and reducing the load on the primary node. This feature is particularly important in read-heavy applications.
+
+2. **Exponential Backoff to Prevent Storm** - A strategy used to prevent connection storms by progressively increasing the wait time between retries when attempting to reconnect to a Valkey server. This helps to reduce the load on the server during topology updates, periods of high demand or network instability.
+
+3. **Valkey Version Compatibility** - Indicates which versions of Valkey the client is compatible with. This is crucial for ensuring that the client can leverage the latest features and improvements in the Valkey server.
+
+4. **PubSub State Restoration** - The ability to restore the state of Pub/Sub (publish/subscribe) channels after a client reconnects. This feature ensures that clients can continue receiving messages after disconnections or topology updates such as adding or removing shards, for both legacy Pub/Sub and sharded Pub/Sub. The client will automatically resubscribe the connections to the new node. The advantage is that the application code is simplified, and doesn’t have to take care of resubscribing to new nodes during reconnects.
+
+5. **Cluster Scan** - This feature ensures that the user experience and guarantees for scanning a cluster are identical to those for scanning a single node. The SCAN function operates as a cursor-based iterator. With each command, the server provides an updated cursor, which must be used as the cursor argument in subsequent calls. A complete iteration with SCAN retrieves all elements present in the collection from start to finish. If an element exists in the collection at the beginning and remains until the end of the iteration, SCAN will return it. Conversely, any element removed before the iteration begins and not re-added during the process will not be returned by SCAN. A client supporting this feature ensures the scan iterator remains valid even during failovers or cluster scaling (in or out) during the SCAN operation. This is achieved by storing metadata of the scanned slots and nodes within the GLIDE SCAN cursor and continuously updating the cluster topology.
```

Comment:
not sure both '_A complete iteration with SCAN retrieves all elements present in the collection from start to finish._' and '_If an element exists in the collection at the beginning and remains until the end of the iteration, SCAN will return it._' are needed, I think maybe one mention is enough.

Reply:
What do you suggest?

Reply:
Is a "GLIDE SCAN" cursor required for this feature, even for clients other than GLIDE?

Drop glide from this description?

Reply:
Other clients use different methods, some dont offer any impl, snd other usually an iterator-like objects.
It is not really an iterator, and its encapsulate complicated behavior from user and led to hard debugging.
Delt with a reach out of user about one of java scan and it was very different than you expect.
We wanted something that act the same as general scan, using cursor without a "state" with side effects.
We want to make usage of cluster as transparent as possible.
We will introduce iterator as well as an extra feature, but not as the general cluster scan.

Reply:
I know what it is. Just don't use "glide" in the general description of a feature. Use it in the description of glide.

Reply:
I agree, will be removed